cmd/sovereignnode/dataCodec/interface.go

```diff
@@ -29,6 +25,7 @@ type OperationDataEncoder interface {
 
 // SovereignDataCodec is the interface for serializing/deserializing data
 type SovereignDataCodec interface {
+	abiSerializer.AbiSerializer
```

Comment:
This sovereign data codec should only have the responsibility of working with our custom structs to serialize/deserialize.

It should only receive the `Serializer` interface, not embed it and extend it.

Also, you added it here just for testing purposes, which is not ok.

Please revert these changes.

If you really need the serializer, you should have in runTypeComps